src/main/java/org/apache/pulsar/manager/service/impl/GithubLoginServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.pulsar.manager.service.impl;
+
+import com.google.common.collect.Maps;
+import com.google.gson.Gson;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.manager.entity.GithubAuthEntity;
+import org.apache.pulsar.manager.entity.GithubUserInfoEntity;
+import org.apache.pulsar.manager.entity.UserInfoEntity;
+import org.apache.pulsar.manager.service.ThirdLoginService;
+import org.apache.pulsar.manager.utils.HttpUtil;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Map;
+
+/**
+ * Github login for get user information.
+ */
+@Slf4j
+@Service
+public class GithubLoginServiceImpl implements ThirdLoginService {
+
+    @Value("${github.client.id}")
+    private String githubClientId;
+
+    @Value("${github.client.secret}")
+    private String githubClientSecret;
```

Comment:
Can we load the secret from system environment?

Reply:
Yes, you can now load the configuration from the environment variable in the following way:

```
export GITHUB_CLIENT_SECRET=client-secret
export GITHUB_CLIENT_ID=client-id
./pulsar-manager/bin/pulsar-manager --github.client.id=$GITHUB_CLIENT_ID --github.client.secret=$GITHUB_CLIENT_ID
```